src/queue.jl

```diff
@@ -1,38 +1,119 @@
 # Utilities for working with HSA queues
 
-"""
-An HSA queue.
-Each queue is uniquely associated with an agent.
-"""
-mutable struct HSAQueue
-    agent::HSAAgent
-    queue::Ref{Ptr{HSA.Queue}}
-    active::Bool
+@static if VERSION >= v"1.7-"
+    """
+    An HSA queue.
+    Each queue is uniquely associated with an agent.
+    """
+    @eval mutable struct HSAQueue
```

Comment:
Why do we need the eval here, but not below? To avoid the expansion of `@atomic`?

Reply:
Yeah, vestige of pre-`@static`. Will remove.